tests/testthat/test-run-nmtran.R

```diff
@@ -0,0 +1,217 @@
+context("testing NM-TRAN")
+
+# define constants
+MODEL_DIR_BBI <- file.path(dirname(ABS_MODEL_DIR), "test-nmtran-models")
+NONMEM_DIR <- Sys.getenv("BBR_TESTS_NONMEM_DIR", "/opt/NONMEM")
+REQ_NONMEM_VERSIONS <- c("nm73gf", "nm74gf", "nm75", Sys.getenv("BBR_TESTS_NONMEM_VERSION"))
+REQ_NONMEM_VERSIONS <- unique(REQ_NONMEM_VERSIONS[REQ_NONMEM_VERSIONS != ""])
+
+# Don't assume NONMEM is available if not on Metworx.
+if (Sys.getenv("METWORX_VERSION") == "" || Sys.getenv("SKIP_BBI_TEST") == "true") {
+  skip("test-workflow-bbi only runs on Metworx because it needs NONMEM installed")
+}
+skip_long_tests("skipping long-running bbi workflow tests")
+
+
+# Skip if required NONMEM versions are not installed
+if(!all(REQ_NONMEM_VERSIONS %in% basename(fs::dir_ls(NONMEM_DIR)))){
+  req_nm_ver_txt <- paste(REQ_NONMEM_VERSIONS, collapse = ", ")
+  skip(glue("test-run-nmtran-bbi only requires the following NONMEM versions {req_nm_ver_txt}"))
+}
+
+
+# cleanup function
+cleanup_bbi <- function(.recreate_dir = FALSE) {
+  if (fs::dir_exists(MODEL_DIR_BBI)) fs::dir_delete(MODEL_DIR_BBI)
+  if (isTRUE(.recreate_dir)) fs::dir_create(MODEL_DIR_BBI)
+}
+cleanup_bbi(.recreate_dir = TRUE)
+
+
+# set options and run tests
+withr::with_options(list(
+  bbr.bbi_exe_path = read_bbi_path(),
+  bbr.verbose = FALSE), {
+
+    # cleanup when done
+    on.exit({
+      Sys.sleep(3) # wait for some NONMEM mess to delete itself
```

Comment:
Is this sleep leftover cruft from `test-workflow-bbi.R`? I can't think of why it's necessary here, and the tests complete fine on my end without it, and three seconds faster.

Reply:
Oh the sleep call specifically? Yeah that's a fair point hadnt really thought about it, but since we're not actually executing any models we shouldn't need that.

Reply:
> Oh the sleep call specifically?

Yes, I was referring to the sleep call when I said "Is this sleep leftover [...]".

Reply:
Skimmed past the most important word ('sleep') there :)